src/services/bundleAnalysis/useBundleTrendData.tsx

```diff
@@ -0,0 +1,234 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import { MissingHeadReportSchema } from 'services/comparison'
+import {
+  RepoNotFoundErrorSchema,
+  RepoOwnerNotActivatedErrorSchema,
+} from 'services/repo'
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+import A from 'ui/A'
+
+export const BUNDLE_TREND_INTERVALS = [
+  'INTERVAL_1_DAY',
+  'INTERVAL_7_DAY',
+  'INTERVAL_30_DAY',
+] as const
+
+export const BUNDLE_TREND_REPORT_TYPES = [
+  'REPORT_SIZE',
+  'JAVASCRIPT_SIZE',
+  'STYLESHEET_SIZE',
+  'FONT_SIZE',
+  'IMAGE_SIZE',
+  'ASSET_SIZE',
+] as const
+
+const MeasurementSchema = z.object({
+  timestamp: z.string(),
+  avg: z.number().nullable(),
+})
+
+const BundleSchema = z.object({
+  measurements: z
+    .array(
+      z.object({
+        __typename: z.literal('BundleAnalysisMeasurements'),
+        assetType: z.enum(BUNDLE_TREND_REPORT_TYPES),
+        measurements: z.array(MeasurementSchema).nullable(),
+      })
+    )
+    .nullable(),
+})
+
+const BundleReportSchema = z.object({
+  __typename: z.literal('BundleAnalysisReport'),
+  bundle: BundleSchema.nullable(),
+})
+
+const BranchSchema = z.object({
+  head: z
+    .object({
+      bundleAnalysisReport: z.discriminatedUnion('__typename', [
+        BundleReportSchema,
+        MissingHeadReportSchema,
+      ]),
+    })
+    .nullable(),
+})
+
+const RepositorySchema = z.object({
+  __typename: z.literal('Repository'),
+  branch: BranchSchema.nullable(),
+})
+
+const RequestSchema = z.object({
+  owner: z
+    .object({
+      repository: z.discriminatedUnion('__typename', [
+        RepositorySchema,
+        RepoNotFoundErrorSchema,
+        RepoOwnerNotActivatedErrorSchema,
+      ]),
+    })
+    .nullable(),
+})
+
+const query = `
+query GetBundleTrend(
+  $owner: String!
+  $repo: String!
+  $branch: String!
+  $bundle: String!
+  $interval: MeasurementInterval!
+  $before: DateTime!
+  $after: DateTime!
+  $filters: BundleAnalysisMeasurementsSetFilters
+) {
+  owner(username: $owner) {
+    repository(name: $repo) {
+      __typename
+      ... on Repository {
+        branch(name: $branch) {
+          head {
+            bundleAnalysisReport {
+              __typename
+              ... on BundleAnalysisReport {
+                bundle(name: $bundle) {
+                  measurements(
+                    interval: $interval
+                    before: $before
+                    after: $after
+                    filters: $filters
+                  ) {
+                    __typename
+                    ... on BundleAnalysisMeasurements {
```

Comment:
are there other types in this list that we should handle?

Reply:
Whoops, my playground was messing up, it's not actually a union type so i'm just gonna remove this, but to answer your question more specifically, there are no other types 😅